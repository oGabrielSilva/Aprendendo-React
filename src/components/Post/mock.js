export const mockPost = {
	posts: [
	{
		id: 1,
		title: 'Title 1',
		body: 'Body 1',
		cover: 'static/logo1.svg'
	},
	{
		id: 2,
		title: 'Title 2',
		body: 'Body 2',
		cover: 'static/logo2.svg'
	},
	{
		id: 3,
		title: 'Title 3',
		body: 'Body 3',
		cover: 'static/logo3.svg'
	}
	]
};
