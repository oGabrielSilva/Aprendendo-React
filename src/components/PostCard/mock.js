export const mockPostCard = {
	post: {
		title: 'Title 1',
		body: 'Body 1',
		id: 1,
		cover: 'static/logo.svg'
	}
};
