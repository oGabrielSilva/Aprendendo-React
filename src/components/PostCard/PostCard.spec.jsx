import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { mockPostCard as mock } from './mock';

const props = mock

describe('<PostCard />', () => {
	it('should render PostCard correctly', () => {
		render(<PostCard { ...props } />);

		expect(screen.getByRole('img', { name: /title 1/i }))
			.toHaveAttribute('src', props.post.cover);
		expect(screen.getByRole('heading', { name: /title 1/i }))
			.toBeInTheDocument();
		expect(screen.getByText(props.post.body)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<PostCard { ...props } />);
		expect(container.firstChild).toMatchSnapshot();
	});
});