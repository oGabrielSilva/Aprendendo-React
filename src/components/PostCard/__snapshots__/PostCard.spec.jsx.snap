// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="static/logo.svg"
  />
  <div
    class="post-content"
  >
    <h1>
      1 Title 1
    </h1>
    <p>
      <span>
        Body 1
      </span>
    </p>
  </div>
</div>
`;
